src/run/serve.rs

```diff
@@ -20,19 +15,44 @@ async fn serve_default_index() -> impl Responder {
         .body(content)
 }
 
+/// Create the default `index.html` if the user didn't provide one.
+fn default_index(bin_target: &BinTarget) -> &'static str {
+    let template = include_str!(concat!(
+        env!("CARGO_MANIFEST_DIR"),
+        "/assets/web/index.html"
+    ));
+
+    // Insert correct path to JS bindings
+    let index = template.replace(
+        "./build/bevy_app.js",
+        format!("./build/{}.js", bin_target.bin_name).as_str(),
+    );
+
+    // Only static strings can be served in the web app,
+    // so we leak the string memory to convert it to a static reference.
+    // PERF: This is assumed to be used only once and is needed for the rest of the app running
+    // time, making the memory leak acceptable.
+    Box::leak(index.into_boxed_str())
+}
+
 /// Launch a web server running the Bevy app.
-pub(crate) fn serve(port: u16, profile: &str) -> anyhow::Result<()> {
-    let profile = profile.to_string();
+pub(crate) fn serve(bin_target: BinTarget, port: u16) -> anyhow::Result<()> {
+    let index_html = default_index(&bin_target);
 
     rt::System::new().block_on(
         HttpServer::new(move || {
             let mut app = App::new();
 
             // Serve the build artifacts at the `/build/*` route
-            // A custom `index.html` will have to call `/build/bevy_app.js`
+            // A custom `index.html` will have to call `/build/{bin_name}.js`
             app = app.service(
-                actix_files::Files::new("/build", wasm_bindgen::get_target_folder(&profile))
-                    .path_filter(|path, _| wasm_bindgen::is_bindgen_artifact(path)),
+                actix_files::Files::new("/build", bin_target.artifact_directory.clone())
+                    // This potentially includes artifacts which we will not need,
+                    // but we can't add the bin name to the check due to lifetime requirements
+                    .path_filter(|path, _| {
+                        path.extension().is_some_and(|ext| ext == "js")
+                            || path.extension().is_some_and(|ext| ext == "wasm")
+                    }),
```

Comment:
I don't know the exact check here, but you can do something like this:

```suggestion
                actix_files::Files::new("/build", bin_target.artifact_directory.clone())
                    let bin_target = bin_target.clone();
                    .path_filter(move |path, _| {
                        path.file_name() == Some(bin_target.bin_name.as_ref())
                            || path.extension().is_some_and(|ext| ext == "js")
                            || path.extension().is_some_and(|ext| ext == "wasm")
                    }),
```

Reply:
I'll be honest, I totally thought this wasn't going to work, because I tried several different things and I was under the assumption that the references needed to be static.

But it totally worked! I adjusted it a bit further so that all relevant files are included

Reply:
The `BinTarget` type is static, it is an owned value that contains no references, so there's no lifetime constraints on it. The problem you had is:

* You didn't mark the `path_filter` closure as move, so it had references to the outer function when you referenced `bin_target`. These references aren't static.
* The closure needs to be called multiple times (i.e. once per request), so you need to clone bin_target outside. This makes the closure `Fn` instead of `FnOnce`.